client-spark/spark-2/src/main/scala/org/apache/spark/shuffle/celeborn/CelebornShuffleFallbackPolicyRunner.scala

```diff
@@ -67,4 +72,15 @@ class CelebornShuffleFallbackPolicyRunner(conf: CelebornConf) extends Logging {
     }
     resp.isAvailable
   }
+
+  /**
+   * If celeborn cluster has no available workers for given shuffle id, fallback to external shuffle.
+   *
+   * @return if celeborn cluster has available workers for given shuffle id.
+   */
+  def checkAvailableWorkers(shuffleId: Int, lifecycleManager: LifecycleManager): Boolean = {
```

Comment:
Hi @SteNicholas , thanks for this PR! However, IMO, `LifecycleManager` should ask `Master` whether there are available workers in the cluster. `LifecycleManager` does not know the cluster status, because it's in Driver of spark or JobMaster of flink, see https://celeborn.apache.org/docs/latest/developers/overview/#components.

So I think we can do this by two ways:
1. Add a new message like `CheckAvailableWorkers`, `LifecycleManager` sends this message to `Master`, `Master` checks by calling `Master#workersAvailable` and responds.
2. Extend the existing `CheckQuota` message, other logic stay the same with the above method, this is simpler.